tests/test_utils.py

```
@@ -0,0 +1,139 @@
+
+                assert "File type: .xyz" in info or "no additional information" in info
+
+    def test_get_binary_file_info_new_vs_modified(self):
+        """Test detecting new vs modified binary files."""
+        with patch("git_commitai.run_command") as mock_run:
+            # Test new file
+            def side_effect_new(cmd, check=True):
+                if "cat-file -s" in cmd:
+                    return "1024"
+                elif "cat-file -e HEAD:" in cmd:
+                    if check:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)